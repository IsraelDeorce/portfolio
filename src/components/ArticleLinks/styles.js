import styled from 'styled-components'

export const ArticleLinks = styled.ul`
  display: flex;
  flex-direction: column;
  padding-left: 1em;
  & > * {
    color: ${({ theme }) => theme.c_brackets};
    &:hover {
      color: ${({ theme }) => theme.c_varType};
    }
    padding-bottom: 1em;
  }
`
