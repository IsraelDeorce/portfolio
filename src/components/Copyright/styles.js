import styled from 'styled-components'

export const Copyright = styled.div`
  display: flex;
  justify-content: center;
`
