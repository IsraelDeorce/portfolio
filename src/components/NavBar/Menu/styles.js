import styled, { css } from 'styled-components'

export const Menu = styled.div`
  flex: 1;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  & > *:not(:last-child) {
    margin-right: 0.75em;
  }
  ${({ isOpen }) => isOpen && css`
    flex-direction: column-reverse;;
    justify-content: flex-end;
    align-items: flex-start;
    background-color: ${({ theme }) => theme.c_menu};
    height: 100vh;
    width: 75vw;
    position: absolute;
    bottom: 0;
    left: 0;
    top: 0;
    box-shadow: blue;
    z-index: 2;
    transition: transform 0.3s ease-in-out;
    box-shadow: 0 5px 30px ${({ theme }) => theme.c_tab};
    padding-left: 2em;
  `}
`

/*
opacity: 0;
-webkit-transform: translate3d(-100%,0,0);
transition-property: opacity,visibility,transform,-webkit-transform;
transition-timing-function: ease-in-out;
transform: translate3d(-100%,0,0);
transition-duration: .25s;
*/


export const UserPreferences = styled.div`
  display: flex;
  align-items: center;
  & > *:not(:last-child) {
    margin-right: 0.75em;
  }
  @media (max-width: 992px) {
    display: none;
    ${({ isOpen }) => isOpen && css`
      display: flex;
      border-bottom: 2px solid gray;
      padding: 0.35em 0;
      margin-bottom: 0.35em;
    `};
  }
`

export const LanguageSelector = styled.div`
  display: flex;
  align-items: center;
  cursor: pointer;
`

export const LanguageCode = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`