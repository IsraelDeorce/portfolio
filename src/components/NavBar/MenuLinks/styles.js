import styled, { css } from 'styled-components'

export const MenuLinks = styled.div`
  display: none
  & > * {
    cursor: pointer;
  }
  & > *:hover {
    color: ${({ theme }) => theme.c_varType};
  }
  @media (min-width: 992px) {
    display: flex;
    & > *:not(:last-child) {
      margin-right: 2em;
    }
  }
  @media (max-width: 992px) {
    display: none;
    
    ${({ isMenuOpen }) => isMenuOpen && css`
      display: flex;
      flex-direction: column;
      & > * {
        margin: .75em 0;
      }
    `}
  }
`
