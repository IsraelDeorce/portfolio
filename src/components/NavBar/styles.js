import styled from 'styled-components'

export const NavBar = styled.nav`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 1;
  display: flex;
  align-items: center;
  font-weight: 700;
  background-color: ${({ theme }) => theme.c_menu};
  color: ${({ theme }) => theme.c_brackets};
  padding: 0 1.5em;
  min-height: 60px;
  & > *:not(:last-child) {
    margin-right: 0.75em;
  }
`
export const Hamburguer = styled.span`
  display: flex;
  align-items: center;
  cursor: pointer;
  flex-shrink: 0;
  &:hover {
    color: ${({ theme }) => theme.c_varType};
  }
  @media (min-width: 992px) {
    display: none;
  }
`

export const Menu = styled.div`
  flex: 1;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  & > *:not(:last-child) {
    margin-right: 0.75em;
  }
  @media (max-width: 992px) {
    display: none;
  }
`
