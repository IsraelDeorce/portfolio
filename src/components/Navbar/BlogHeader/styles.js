import styled from 'styled-components'

export const BlogHeader = styled.nav`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 1;
  display: flex;
  align-items: center;
  font-weight: 700;
  background-color: ${({ theme }) => theme.c_menu};
  color: ${({ theme }) => theme.c_brackets};
  padding: 0 1.5em;
  min-height: 60px;
  & > *:not(:last-child) {
    margin-right: 0.75em;
  }
  @media (min-width: 992px) {
    display: none;
  }
`
export const Hamburguer = styled.span`
  display: flex;
  align-items: center;
  cursor: pointer;
  flex-shrink: 0;
  &:hover {
    color: ${({ theme }) => theme.c_varType};
  }
`
