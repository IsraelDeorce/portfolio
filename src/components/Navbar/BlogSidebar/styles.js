import styled from 'styled-components'

export const Background = styled.div`
  opacity: 0.7;
  background-color: ${({ theme }) => theme.c_background};
  position:fixed;
  width:100%;
  height:100%;
  top:0px;
  left:0px;
  @media (min-width: 992px) {
    display: none;
  }
`

export const BlogSidebar = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  display: flex;
  flex-direction: column;
  background-color: ${({ theme }) => theme.c_menu};
  height: 100vh;
  width: 65vw;
  z-index: 2;
  box-shadow: 0 5px 30px ${({ theme }) => theme.c_tab};
  padding: 0 1em 1em 1em;
  & > * {
    width: 100%;
  }
  & > :last-child {
    align-self: center;
  }
  @media (min-width: 992px) {
    position: static;
    top: auto;
    left: auto;
    right: auto;
    height: auto;
    width: 285px;
    min-width: 285px;
    box-shadow: none;
  }
`

export const Header = styled.div`
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  min-height: 60px;
  border-bottom: 2px solid gray;
  & > *:not(:last-child) {
    padding: 0.35em;
  }
`

export const Preferences = styled.div`
  display: flex;
  align-items: center;
  & > *:not(:last-child) {
    margin-right: 0.75em;
  }
`
