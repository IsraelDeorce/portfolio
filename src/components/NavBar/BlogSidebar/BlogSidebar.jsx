import { useEffect } from 'react'
import { useSelector, useDispatch } from 'react-redux'
import { Link } from 'react-router-dom'
import 'react-toggle/style.css'
import { useWindowSize } from '../../../utils/hooks'
import { TitledLogo, ThemeToggle, LanguageSelector, ArticleLinks, UIButton } from '../../'
import * as S from './styles'

function BlogSidebar() {
  const { isBlogSidebarOpen } = useSelector((state) => state.session)
  const [_height, width] = useWindowSize()
  const dispatch = useDispatch()

  useEffect(() => {
    if (isBlogSidebarOpen) {
      document.body.style.overflow = 'hidden'
      if (width >= 992) dispatch.session.update({ isBlogSidebarOpen: false })
    } else {
      document.body.style.overflow = 'visible'
    }
  }, [isBlogSidebarOpen, width])

  const closeBlogSidebar = () => { dispatch.session.update({ isBlogSidebarOpen: false }) }

  if (!isBlogSidebarOpen && width < 992) return null
  
  return (
    <>
      <S.Background onClick={closeBlogSidebar} />
      <S.BlogSidebar>
        <S.Header>
          <Link to='/'>
            <TitledLogo />
          </Link>
          <S.Preferences>
            <ThemeToggle />
            <LanguageSelector />
          </S.Preferences>
        </S.Header>
        <ArticleLinks onClick={closeBlogSidebar} />
      </S.BlogSidebar>
    </>
  )
}

export default BlogSidebar
