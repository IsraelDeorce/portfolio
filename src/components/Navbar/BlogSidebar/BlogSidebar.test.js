import { useSelector, useDispatch } from 'react-redux'
import { HashRouter as Router } from 'react-router-dom'
import BlogSidebar from './BlogSidebar'
import * as S from './styles'

describe('<BlogSidebar/>', () => {
  const session = useSelector((state) => state.session)
  const dispatch = useDispatch()

  beforeEach(() => {
    session.isBlogSidebarOpen = false
    global.innerWidth = 1000
  })
  afterEach(() => { jest.clearAllMocks() })

  it('does not render when isBlogSidebarOpen is false and width is smaller than 992 pixels', () => {
    session.isBlogSidebarOpen = false
    global.innerWidth = 991
    const shallowed = mount(<BlogSidebar />)
    shallowed.update()
    expect(shallowed.isEmptyRender()).toBe(true)
  })

  it('closes the BlogSidebar when the user clicks on the background', () => {
    shallow(<BlogSidebar />).find(S.Background).simulate('click')
    expect(dispatch.session.update).toHaveBeenCalledTimes(1)
    expect(dispatch.session.update).toHaveBeenCalledWith({ isBlogSidebarOpen: false })
  })

  it('renders the titled logo as a link that redirects to the default page', () => {
    const Link = shallow(<BlogSidebar />).find('Link')
    expect(Link.prop('to')).toBe('/')
    expect(Link.find('TitledLogo')).toHaveLength(1)
  })

  it('closes the BlogSidebar when the user clicks on the article links', () => {
    shallow(<BlogSidebar />).find('ArticleLinks').simulate('click')
    expect(dispatch.session.update).toHaveBeenCalledTimes(1)
    expect(dispatch.session.update).toHaveBeenCalledWith({ isBlogSidebarOpen: false })
  })

  it('closes the BlogSidebar when it was open and the screen width is 992 pixels wide', () => {
    session.isBlogSidebarOpen = true
    global.innerWidth = 992
    mount(<Router><BlogSidebar /></Router>)
    expect(dispatch.session.update).toHaveBeenCalledTimes(1)
    expect(dispatch.session.update).toHaveBeenCalledWith({ isBlogSidebarOpen: false })
  })

  it('closes the BlogSidebar when it was open and the screen width is wider than 992 pixels', () => {
    session.isBlogSidebarOpen = true
    global.innerWidth = 993
    mount(<Router><BlogSidebar /></Router>)
    expect(dispatch.session.update).toHaveBeenCalledTimes(1)
    expect(dispatch.session.update).toHaveBeenCalledWith({ isBlogSidebarOpen: false })
  })

  it('does not close the BlogSidebar when the screen width is smaller than 992 pixels', () => {
    global.innerWidth = 991
    mount(<Router><BlogSidebar /></Router>)
    expect(dispatch.session.update).toHaveBeenCalledTimes(0)
  })

  it('hides the screen overflow when the BlogSidebar is open', () => {
    session.isBlogSidebarOpen = true
    mount(<Router><BlogSidebar /></Router>)
    expect(document.body.style.overflow).toBe('hidden')
  })

  it('returns the screen overflow when the BlogSidebar is closed', () => {
    session.isBlogSidebarOpen = false
    mount(<Router><BlogSidebar /></Router>)
    expect(document.body.style.overflow).toBe('visible')
  })

  describe('Snapshot', () => {
    it('renders without crashing using shallow', () => {
      expect(shallow(<BlogSidebar />)).toMatchSnapshot()
    })
  })
})
