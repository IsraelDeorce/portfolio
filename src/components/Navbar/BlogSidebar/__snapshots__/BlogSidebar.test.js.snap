// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogSidebar/> Snapshot renders without crashing using shallow 1`] = `
<Fragment>
  <styled.div
    onClick={[Function]}
  />
  <styled.div>
    <styled.div>
      <Link
        to="/"
      >
        <TitledLogo />
      </Link>
      <styled.div>
        <ThemeToggle />
        <LanguageSelector />
      </styled.div>
    </styled.div>
    <ArticleLinks
      onClick={[Function]}
    />
  </styled.div>
</Fragment>
`;
